### PR TITLE
Make COCOTB_REDUCED_LOG_FMT enabled by default

### PR DESCRIPTION
**Pull request by ktbarrett**
xref #2564. Turns `COCOTB_REDUCED_LOG_FMT` from an existential flag into a boolean option. The option defaults to `1`, so the reduced format is the new standard format, and the old full format can be selected by setting the variable to `0`.

### CODE REVIEW

File: cocotb/log.py [hunk 1]
```diff
@@ -40,10 +40,10 @@
 
 import cocotb.ANSI as ANSI
 
-if "COCOTB_REDUCED_LOG_FMT" in os.environ:
-    _suppress = True
-else:
-    _suppress = False
+try:
+    _suppress = int(os.environ.get("COCOTB_REDUCED_LOG_FMT", "1"))
```
Comment (cmarqu) [inline cocotb/log.py:44]:
```suggestion
    _suppress = int(os.getenv("COCOTB_REDUCED_LOG_FMT", default="1"))
```
The existing codebase leans toward `getenv` I think.

File: documentation/source/newsfragments/2564.change.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Make :envvar:`COCOTB_REDUCED_LOG_FMT` enabled by default.
```
Comment (cmarqu) [inline documentation/source/newsfragments/2564.change.rst:1]:
```suggestion
Shorter log lines (configurable with :envvar:`COCOTB_REDUCED_LOG_FMT`) are now the default.
```

**Review by cmarqu — APPROVED** [review]

File: cocotb/log.py (hunk 1 above)
Comment (imphil) [reply cocotb/log.py:44]: No strong feelings wither way, but I would also go for `os.environ.get()` :-)

File: documentation/source/newsfragments/2564.change.rst (hunk 2 above)
Comment (imphil) [reply documentation/source/newsfragments/2564.change.rst:1]: Can we add a second sentence: "For wider log output, similar to previous cocotb releases, set the COCOTB_REDUCED_LOG_FMT environment variable to 0."

**Review by themperek — APPROVED** [review]

**Review by imphil — APPROVED** [review]

File: cocotb/log.py (hunk 1 above)
Comment (ktbarrett) [reply cocotb/log.py:44]: And this is why "There should be one &mdash; and preferably only one &mdash; obvious way to do it." =)

File: documentation/source/newsfragments/2564.change.rst (hunk 2 above)
Comment (ktbarrett) [reply documentation/source/newsfragments/2564.change.rst:1]:
```suggestion
Shorter log lines (configurable with :envvar:`COCOTB_REDUCED_LOG_FMT`) are now the default. For wider log output, similar to previous cocotb releases, set the :envvar:`COCOTB_REDUCED_LOG_FMT` environment variable to ``0``.
```

**Pull request merged**